grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java

```diff
@@ -312,13 +312,33 @@ public void reusingResources4() {
         assertEquals(answers.size(), 1);
     }
 
-    @Ignore
-    @Test //Expected result: Two answers with obtained only if checked the instance type compatible with the one specified with rule body.
+    @Test //Expected result: Two answers obtained only if the rule query containing sub type is correctly executed.
     public void instanceTypeHierarchyRespected(){
         QueryBuilder qb = testSet18.graph().graql().infer(true);
-        String queryString1 = "match $x isa entity1;(role1: $x, role2: $y) isa relation1;";
-        QueryAnswers answers = queryAnswers(qb.parse(queryString1));
+        String queryString = "match " +
+                "$x isa entity1;" +
+                "$y isa entity1;" +
+                "(role1: $x, role2: $y) isa relation1;";
+        String queryString2 = queryString + "$y has name 'a';";
+        QueryAnswers answers = queryAnswers(qb.parse(queryString));
         assertEquals(answers.size(), 2);
+        QueryAnswers answers2 = queryAnswers(qb.parse(queryString2));
+        assertEquals(answers2.size(), 2);
+    }
+
+    @Test //Expected result: Single answer obtained only if the rule query containing super type is correctly executed.
```

Comment:
Putting these types of comments on the tests above would help a lot in debugging I think